File: clients/java/Mutation.java
import com.sun.jna.Pointer;

///Represents a mutation of Plant.
class Mutation
{
  Pointer ptr;
  int ID;
  int iteration;
  public Mutation(Pointer p)
  {
    ptr = p;
    ID = Client.INSTANCE.mutationGetId(ptr);
    iteration = BaseAI.iteration;
  }
  boolean validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.mutations.length; i++)
    {
      if(BaseAI.mutations[i].ID == ID)
      {
        ptr = BaseAI.mutations[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

    //commands


    //getters

  ///Unique Identifier
  public int getId()
  {
    validify();
    return Client.INSTANCE.mutationGetId(ptr);
  }
  ///The name of this mutation of Plant.
  public String getName()
  {
    validify();
    return Client.INSTANCE.mutationGetName(ptr);
  }
  ///The Mutation specific id representing this mutation of Plant.
  public int getMutation()
  {
    validify();
    return Client.INSTANCE.mutationGetMutation(ptr);
  }
  ///The spore cost to spawn this Plant mutation into the game.
  public int getSpores()
  {
    validify();
    return Client.INSTANCE.mutationGetSpores(ptr);
  }
  ///The maximum number of times the Plant can attack.
  public int getMaxAttacks()
  {
    validify();
    return Client.INSTANCE.mutationGetMaxAttacks(ptr);
  }
  ///The maximum amount of this health this Plant can have
  public int getMaxHealth()
  {
    validify();
    return Client.INSTANCE.mutationGetMaxHealth(ptr);
  }
  ///The maximum number of moves this Plant can move.
  public int getMaxMovement()
  {
    validify();
    return Client.INSTANCE.mutationGetMaxMovement(ptr);
  }
  ///The range of this Plant's attack.
  public int getRange()
  {
    validify();
    return Client.INSTANCE.mutationGetRange(ptr);
  }
  ///The minimum strength of this mutation's attack/heal/buff
  public int getMinStrength()
  {
    validify();
    return Client.INSTANCE.mutationGetMinStrength(ptr);
  }
  ///The base strength of this mutation's attack/heal/buff
  public int getBaseStrength()
  {
    validify();
    return Client.INSTANCE.mutationGetBaseStrength(ptr);
  }
  ///The power of this plant's attack/heal/buff
  public int getMaxStrength()
  {
    validify();
    return Client.INSTANCE.mutationGetMaxStrength(ptr);
  }
  ///The power of this Plant mutation's attack.
  public int getMaxStorage()
  {
    validify();
    return Client.INSTANCE.mutationGetMaxStorage(ptr);
  }

}

File: clients/csharp/Mutation.cs
using System;
using System.Runtime.InteropServices;

/// <summary>
/// Represents a mutation of Plant.
/// </summary>
public class Mutation
{
  public IntPtr ptr;
  protected int ID;
  protected int iteration;

  public Mutation()
  {
  }

  public Mutation(IntPtr p)
  {
    ptr = p;
    ID = Client.mutationGetId(ptr);
    iteration = BaseAI.iteration;
  }

  public bool validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.mutations.Length; i++)
    {
      if(BaseAI.mutations[i].ID == ID)
      {
        ptr = BaseAI.mutations[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

  #region Commands
  #endregion

  #region Getters
  /// <summary>
  /// Unique Identifier
  /// </summary>
  public int Id
  {
    get
    {
      validify();
      int value = Client.mutationGetId(ptr);
      return value;
    }
  }

  /// <summary>
  /// The name of this mutation of Plant.
  /// </summary>
  public string Name
  {
    get
    {
      validify();
      IntPtr value = Client.mutationGetName(ptr);
      return Marshal.PtrToStringAuto(value);
    }
  }

  /// <summary>
  /// The mutation of this Plant. This value is unique for all Mutations.
  /// </summary>
  public int Type
  {
    get
    {
      validify();
      int value = Client.mutationGetType(ptr);
      return value;
    }
  }

  /// <summary>
  /// The current amount of radiation this Plant has.
  /// </summary>
  public int Spores
  {
    get
    {
      validify();
      int value = Client.mutationGetSpores(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum number of times the Plant can radiate.
  /// </summary>
  public int MaxRadiates
  {
    get
    {
      validify();
      int value = Client.mutationGetMaxRadiates(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum amount of radiation this Plant can have before dying.
  /// </summary>
  public int MaxRads
  {
    get
    {
      validify();
      int value = Client.mutationGetMaxRads(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum range this plant can radiate.
  /// </summary>
  public int Range
  {
    get
    {
      validify();
      int value = Client.mutationGetRange(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum number of times this plant can be uprooted.
  /// </summary>
  public int MaxUproots
  {
    get
    {
      validify();
      int value = Client.mutationGetMaxUproots(ptr);
      return value;
    }
  }

  /// <summary>
  /// The minimum power of this plant.
  /// </summary>
  public int MinStrength
  {
    get
    {
      validify();
      int value = Client.mutationGetMinStrength(ptr);
      return value;
    }
  }

  /// <summary>
  /// The base power of this plant.
  /// </summary>
  public int BaseStrength
  {
    get
    {
      validify();
      int value = Client.mutationGetBaseStrength(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum power of this plant.
  /// </summary>
  public int MaxStrength
  {
    get
    {
      validify();
      int value = Client.mutationGetMaxStrength(ptr);
      return value;
    }
  }

  #endregion

  #region Properties
  #endregion
}
